From 1c19492c453426c6b9f7af868e9b24b0734d9617 Mon Sep 17 00:00:00 2001
From: hluwa <hluwa888@gmail.com>
Date: Mon, 16 Aug 2021 10:55:11 +0800
Subject: [PATCH 8/8] strongR-frida: protocol_unexpected_command

---
 src/droidy/droidy-client.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/droidy/droidy-client.vala b/src/droidy/droidy-client.vala
index e784e0d1..95ca0deb 100644
--- a/src/droidy/droidy-client.vala
+++ b/src/droidy/droidy-client.vala
@@ -974,7 +974,7 @@ namespace Frida.Droidy {
 						case "OPEN":
 						case "CLSE":
 						case "WRTE":
-							throw new Error.PROTOCOL ("Unexpected command");
+							break; //throw new Error.PROTOCOL ("Unexpected command");
 
 						default:
 							var length = parse_length (command_or_length);
-- 
2.30.2

