From 3abc151103db6b7ac3c971a5dfc5f90299d3932f Mon Sep 17 00:00:00 2001
From: Ylarod <me@ylarod.cn>
Date: Tue, 18 Jul 2023 15:52:56 +0800
Subject: [PATCH 02/10] Florida: re_frida_server

---
 server/server.vala | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/server/server.vala b/server/server.vala
index 525c145e..f7547819 100644
--- a/server/server.vala
+++ b/server/server.vala
@@ -1,7 +1,7 @@
 namespace Frida.Server {
 	private static Application application;
 
-	private const string DEFAULT_DIRECTORY = "re.frida.server";
+	private static string DEFAULT_DIRECTORY = null;
 	private static bool output_version = false;
 	private static string? listen_address = null;
 	private static string? certpath = null;
@@ -50,6 +50,7 @@ namespace Frida.Server {
 	};
 
 	private static int main (string[] args) {
+		DEFAULT_DIRECTORY = GLib.Uuid.string_random();
 		Environment.init ();
 
 #if DARWIN
-- 
2.39.3 (Apple Git-145)

