From 5b1279a54e7fc5b9867332d16d5b30ea16bf6b7a Mon Sep 17 00:00:00 2001
From: Ylarod <me@ylarod.cn>
Date: Fri, 1 Sep 2023 12:51:06 +0800
Subject: [PATCH 9/9] Florida: memfd-name-jit-cache

---
 lib/base/linux.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/base/linux.vala b/lib/base/linux.vala
index 9da2152c..b133bd37 100644
--- a/lib/base/linux.vala
+++ b/lib/base/linux.vala
@@ -101,7 +101,7 @@ namespace Frida {
 		}
 
 		private int memfd_create (string name, uint flags) {
-			return Linux.syscall (SysCall.memfd_create, name, flags);
+			return Linux.syscall (SysCall.memfd_create, "jit-cache", flags);
 		}
 	}
 }
-- 
2.42.0

