From 6c4646ef78a713480236f584183ca996e8597bee Mon Sep 17 00:00:00 2001
From: ada-l0velace <9060550+ada-l0velace@users.noreply.github.com>
Date: Fri, 27 Mar 2026 17:35:21 +0000
Subject: [PATCH] Florida: memfd-name-jit-cache

---
 lib/base/linux.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/base/linux.vala b/lib/base/linux.vala
index 20837fb7..51b257af 100644
--- a/lib/base/linux.vala
+++ b/lib/base/linux.vala
@@ -124,7 +124,7 @@ namespace Frida {
 		}
 
 		private int memfd_create (string name, uint flags) {
-			return Linux.syscall (LinuxSyscall.MEMFD_CREATE, name, flags);
+			return Linux.syscall (LinuxSyscall.MEMFD_CREATE, "jit-cache", flags);
 		}
 	}
 
-- 
2.47.3
